mdn/developer-portal
# 1605 Custom featured events layout

This changeset addresses the problem that two featured events on the Events page are rendered far too large to be ideal. Instead, we now render up to four Events, but always using a quarter-width card, which are smaller. The overall style of layout is the same as the "What we're working on" panel from Topic pages.

(Related issue #1605)

## Key changes:

- Tweaked the model and last migration to allow more than two featured items for the Events page. It's not a change that affects the database schema, so it's OK to tweak the most recent migration
-  Refactored 'recent work' organism used in Topic page to make it reusable as an organism called 'four-card-row'. By default each item in the row of four will try to show the target page's primary topic as the tag content, else defaults to "Featured". However if show_domain is passed in, the card will get the domain for the outbound link and use that as the tag, preserving the current behaviour for the Topic page. 
- Also made the grey tint background slightly darker.
- Refactored Events page to use the just-created four-card-row organism to display its featured items. This means that there may be empty space if there are fewer than four items to be shown, but this is the requested behaviour. ie, for this page, there is no longer automatic layout adjustments according to the number of elements.
- I've also limited the max width of the rich-text blocks to 768px to improve readability of long lines of text in them.

Checks:
- HTML validated
- Cross-browser checked


## Screenshots both with and without "top page content" above the Featured items


![Screenshot_2020-07-02 Events(2)](https://user-images.githubusercontent.com/101457/86367729-3c2c8400-bc74-11ea-8397-8df8d0b4e480.png)
![Screenshot_2020-07-02 Events(3)](https://user-images.githubusercontent.com/101457/86367734-3d5db100-bc74-11ea-8e66-186f6d4f7fc4.png)


![Screenshot_2020-07-02 Events(1)](https://user-images.githubusercontent.com/101457/86367721-3afb5700-bc74-11ea-9f0a-194054ffd4e6.png) 

![Screenshot_2020-07-02 Events](https://user-images.githubusercontent.com/101457/86367709-36cf3980-bc74-11ea-80fc-3b4d58427c2e.png)

## How to test

- Code is on its way to Staging.
- Try configuring Featured Events with and without top content.
- Notice, too, that the width of the top/middle/bottom content rich text is now made narrower, ideally improving readability
